src/main/java/com/hubspot/imap/protocol/response/tagged/CapabilityResponse.java

```diff
@@ -0,0 +1,33 @@
+package com.hubspot.imap.protocol.response.tagged;
+
+import com.hubspot.imap.protocol.capabilities.Capabilities;
+import com.hubspot.imap.protocol.response.untagged.UntaggedCapabilityResponse;
+
+public interface CapabilityResponse extends TaggedResponse {
+
+  Capabilities getCapabilities();
+
+  class Builder extends TaggedResponse.Builder implements CapabilityResponse {
+    private Capabilities capabilities;
+
+    public CapabilityResponse fromResponse(TaggedResponse input) {
+      capabilities = input.getUntagged().stream()
+          .filter(o -> o instanceof UntaggedCapabilityResponse)
+          .map(o -> ((UntaggedCapabilityResponse) o))
+          .map(UntaggedCapabilityResponse::getCapabilities)
+          .findFirst()
```

Comment:
Do we need to sort first before using `findFirst()`?

Reply:
there will never actually be more than one element here, this is merely a convenience